src/libsystemd/sd-bus/sd-bus.c

```diff
@@ -4142,8 +4142,7 @@ _public_ int sd_bus_path_decode_many(const char *path, const char *path_template
 
                 /* verify the suffixes match */
                 sep = strchrnul(path_pos, '/');
-                if (sep - path_pos < (ssize_t)length ||
-                    strncmp(sep - length, template_pos, length))
+                if (sep - path_pos < (ssize_t) length || !strneq(sep - length, template_pos, length))
```

Comment:
Also here.

Reply:
Ditto what I said earlier.

Reply:
@yuwata That cannot apply here due to the processing of the string being into pieces, as in we are not looking for only a null terminator, but dividing a string by the slashes as well.